### PR TITLE
[gradle-plugin] Read Maven group from GROUP property

### PR DESCRIPTION
**Pull request by douglowder**
## Summary:

The [React Native TV repo](https://github.com/react-native-tvos/react-native-tvos) shares most of the same Android code as the core repo. Beginning in 0.71, it needs to also publish Android Maven artifacts for the `react-android` and `hermes-android` libraries.

In order to avoid conflicts, it needs to publish the artifacts to a different group name. However, `react-native-gradle-plugin` uses a hardcoded group name (`com.facebook.react`).

Solution: read the group name from the existing `GROUP` property in `ReactAndroid/gradle.properties`.

## Changelog:

[Android] [Fixed] - read GROUP name in gradle-plugin dependency code

## Test Plan:

- Android unit tests have been added for the new code and new method in `DependencyUtils.kt`.
- Existing tests should pass
- The new code defaults to the correct group (`com.facebook.react`) so no functional change is expected in the core repo.

### CODE REVIEW

File: packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt [hunk 1]
```diff
@@ -84,9 +84,18 @@ internal object DependencyUtils {
     }
   }
 
+  // Allows react-native-tvos to get its dependencies correctly with its custom Maven group name
+  fun readGroupString(propertiesFile: File): String {
```
Comment (cortinico) [inline packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt:88]: Can we merge `readVersionString` and `readGroupString` into a function that returns a `Pair<String, String>` so we'll be reading from file only once?

File: packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt [hunk 2]
```diff
@@ -84,9 +84,18 @@ internal object DependencyUtils {
...
+    return reactAndroidProperties["GROUP"] as? String ?: defaultGroupString
+  }
+
   fun Project.mavenRepoFromUrl(url: String): MavenArtifactRepository =
       project.repositories.maven { it.url = URI.create(url) }
 
   fun Project.mavenRepoFromURI(uri: URI): MavenArtifactRepository =
       project.repositories.maven { it.url = uri }
+
+  val defaultGroupString = "com.facebook.react"
```
Comment (cortinico) [inline packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt:100]: Let's move this out of this class to the top level and make it a 
```suggestion
  internal const val DEFAULT_GROUP_STRING = "com.facebook.react"
```

**Review by cortinico — APPROVED** [review]
Thanks for sending this 👍 left a couple of nits

File: packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt (hunk 2 above)
Comment (douglowder) [reply packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt]: Done

File: packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt (hunk 1 above)
Comment (douglowder) [reply packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt]: Done

**Comment by douglowder** [comment]
@cortinico just saw that there was a merge conflict with the imports in `ReactPlugin.kt` and fixed that

**Pull request closed without merging**